Ofirvins/next

ofirvin opened the pull request:
Fix XPRT memory leaks.

in cases where XPRT is shutdown when there are requests in queue, xprt will never be released as requests keep reference for xprt but no logic will clear them

ofirvin committed:
svc_rqst.c: Return error when we do not requeue

ofirvin committed:
Do not release xprt on rearm fail

This is incorrect as xioq is still in writeq which can
cause us to release xprt multiple times for same request

ofirvin committed:
Add missing xprt releases

ofirvin committed:
Fix compilation under C23.

C23 adds a boolean type, and thus now errors out on `return false` in a function which returns a pointer.

ffilz commented on ntirpc/rpc/svc.h:
```diff
@@ -178,6 +178,7 @@ typedef struct svc_init_params {
 #define SVC_XPRT_FLAG_UREG		0x0080
 #define SVC_XPRT_TREE_LOCKED		0x0100
 #define SVC_XPRT_FLAG_REMOTE_ADDR_SET	0x0200	/* remote addr was final set */
+#define SVC_XPRT_FLAG_IOQ_WRITING	0x0400	/* xprt is used by svc_ioq_write */
```
This will need to be changed to 0x0800, PR #328 is using 0x0400

ofirvin replied:
thanks for the heads up I saw it, I am waiting for it to be merged and then I will update this

ofirvin replied:
updated

ffilz reviewed (commented):
This all looks credible on first glance, but it's a pretty significant patch set - this may have to wait for V7-dev.1 (and then be backported to V6-stable).

ofirvin committed:
release all xprt non ongoing requests when xprt is destroyed

Currently waiting requests on moment of destruction will never  be freed.
which means XPRT will never be freed as they hold references to it

ffilz commented:
Just flagging as duplicate since these patches are now in PR #332

liorsu commented:
Yep, we closed this PR and moved it to #332 that includes more commits in addition to those ones

liorsu commented:
Sorry for the confusion
